### ISSUE TITLE
cover.jpg

### ISSUE THREAD

**Issue by ozonepilot** [rvolz/BicBucStriim#167]
Love your program. One thing tho', I saw a few books I have that were missing covers. I found some [not with the Calibre operation] but on my own, downloaded them, put them in my Library on my Syno NAS. Opened Calibre on the PC and made sure that it pointed to that cover.jpg. Hiwever, when I log into the library, the "standard" no cover jpg is there, but when I click on the book in my library, it finds the cover I put in there. Why won't the cover.jpg update in the book list but only when I locate and click on the book itself? Thanks !!

**Comment by rvolz** [comment]
BicBucStriim will display what Calibre defines. If there is no cover image defined it will show a *missing link* in the book list, and no image in the details view. If the app displays an old, outdated image in the list, then you can delete that. The thumbnails are stored as  `<bbs>/data/titles/thumb_<title_no>.png`. Simply delete the old image and the thumbnail will be regenerated.

**Comment by ozonepilot** [comment]
Thanks for your reply! I do have a little problem tho’. I’ve got many books without covers, even though I went through Calibre to obtain a few using more search items [for book covers]. Problem is. Because of the amount of covers I’m missing, I’d have to search tem out one by one [a tedious job] which I don’y mond to do. The location you gave me below of locating the covers is good, but I’m trying to determine how to locate a single book cover, as the files are only in some numerical order such as “thumb_1481” How do I match up the title to the cover? 

Appreciate your replies and your program!!

**Comment by rvolz** [comment]
The number in "thumb_1481" is the Calibre ID for the book. One way to match them could be to look at your Calibre library directory, something like this will appear:

>── John Ruskin
>│   └── The Stones of Venice, Volume II (7)
>│       ├── cover.jpg
>│       ├── metadata.opf
>│       ├── The Stones of Venice, Volume II - John Ruskin.epub
>│       ├── The Stones of Venice, Volume II - John Ruskin.mobi
>│       └── The Stones of Venice, Volume II - John Ruskin.pdf
>├── Joseph von Eichendorff
>│   └── Die Glucksritter (4)
>│       ├── cover.jpg
>│       ├── Die Glucksritter - Joseph von Eichendorff.epub
>│       ├── Die Glucksritter - Joseph von Eichendorff.pdf
>│       └── metadata.opf
>├── metadata.db

The number between the braces is the Calibre ID that is used for the thumbnails.

 Alternatively you could look at the URL in your browser if you selected a certain book, eg 

> http://.../bbs/titles/4

The last number is again the Calibre ID.

**Comment by ozonepilot** [comment]
Ah great, looking at the URL, [silly me, I should’ve known better] gave me the results I needed. Still a problem though. I copied over the new “cover.jpg” into my Library and I get these results….when going back to the /bbs URL, and look for the book I just replaced, I still see the old .png file [no name index file], but clicking on the book title, I see the new “cover.jpg” I used to replace the old one. Must I also replace the “thumb_XX.png” as well or are they supposed to regenerate as mentioned on your site? So must I replace both copies, one in my Library as a .jpg and the other in the /bbs/data/titles/.png?

 

I restarted the web server for good measure to see if that would work, but it still gives me the index file [assuming it’s the .png as the old .png file, the cover.jpg is fine. Sorry to be a pest
